### ISSUE TITLE
Add integrated gradients for more interpretability 

### ISSUE THREAD

**Issue by AliAshraf14** [BrikerMan/Kashgari#448]
**Is your feature request related to a problem? Please describe.**
I'm always fond of tools that allow researchers to interpret their models' outputs such as [Grad Cam](https://arxiv.org/pdf/1611.07450.pdf) and [Captum](https://captum.ai/). Unfortunately, there are no similar tools implemented natively in this framework
**Describe the solution you'd like**
Adding integrated gradients to the framework will improve the interpretability of the models built using it, while also holding up to the "Production-ready, Human-ready" standards.
**Describe alternatives you've considered**
I have tried [Captum](https://captum.ai/). Unfortunately, it only works with Pytorch. I also tried to implement the proposed solution, but I ran into some issues such as the solution requires an exact layer-which I was able to solve-, and the solution requires the input to have the same embedding and padding done in the training function-which I was not able to solve-.
**Additional context**
If this feature is successfully added to the framework, it will be grand. But if it's not feasible, any tips with the problems stated above will be very much appreciated while I implement it myself.
